test/test_uploader.py

```diff
@@ -754,6 +754,107 @@ def test_eval_upload_parameter(self):
         self.assertIsInstance(result['quality_analysis'], dict)
         self.assertIsInstance(result['quality_analysis']['focus'], float)
 
+    @unittest.skipUnless(cloudinary.config().api_secret, "requires api_key/api_secret")
+    def test_generate_sprite(self):
+        """Should generate a sprite from all images associated with a tag or from the image urls"""
+        sprite_test_tag = "sprite_test_tag{}".format(SUFFIX)
+        images_quantity_in_sprite = 2
+
+        upload_result_1 = uploader.upload(TEST_IMAGE, tags=[sprite_test_tag, UNIQUE_TAG],
+                                          public_id="sprite_test_tag_1{}".format(SUFFIX))
+        upload_result_2 = uploader.upload(TEST_IMAGE, tags=[sprite_test_tag, UNIQUE_TAG],
+                                          public_id="sprite_test_tag_2{}".format(SUFFIX))
+
+        result = uploader.generate_sprite(tag=sprite_test_tag, tags=[UNIQUE_TAG])
+        self.assertEqual(len(result["image_infos"]), images_quantity_in_sprite)
+
+        urls = [upload_result_1.get('url'), upload_result_2.get('url')]
+        result = uploader.generate_sprite(urls=urls, tags=[UNIQUE_TAG])
+        self.assertEqual(len(result["image_infos"]), images_quantity_in_sprite)
+
+        result = uploader.generate_sprite(sprite_test_tag, transformation={"raw_transformation": "w_100"})
+        self.assertIn("w_100", result["css_url"])
+
+        result = uploader.generate_sprite(sprite_test_tag, format="jpg", width=100)
+        uploader.destroy(result.get("public_id"))
+        self.assertIn("f_jpg,w_100", result["css_url"])
+
+    def test_download_sprite(self):
+        """Should generate signed download url for sprite"""
+        sprite_test_tag = "sprite_tag"
+        url_1 = "https://res.cloudinary.com/demo/image/upload/sample"
+        url_2 = "https://res.cloudinary.com/demo/image/upload/car"
+
+        url_from_tag = uploader.download_generated_sprite(tag=sprite_test_tag)
+        url_from_urls = uploader.download_generated_sprite(urls=[url_1, url_2])
+
+        self.assertTrue(url_from_tag.startswith(
+            "https://api.cloudinary.com/v1_1/" + cloudinary.config().cloud_name + "/image/sprite"))
+        self.assertTrue(url_from_urls.startswith(
+            "https://api.cloudinary.com/v1_1/" + cloudinary.config().cloud_name + "/image/sprite"))
+
+        parameters = parse_qs(urlparse(url_from_tag).query)
+        self.assertEqual(sprite_test_tag, parameters["tag"][0])
+        self.assertEqual("download", parameters["mode"][0])
+        self.assertTrue(parameters["timestamp"])
```

Comment:
```suggestion
        self.assertIn("timestamp", parameters)
```
Looks more appropriate and easier to understand and won't throw `KeyError` if it's not present

This suggestion is relevant in other similar places